File: support_vector_regression.R
#Regression Template
#Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

#Encoding categorical data
# dataset$Age = ifelse(is.na(dataset$Age), 
#                      ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)), 
#                      dataset$Age)
# dataset$Salary = ifelse(is.na(dataset$Salary), 
#                         ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)), 
#                         dataset$Salary)
# 
# dataset$Country = factor(dataset$Country, 
#                          levels = c('France', 'Spain', 'Germany'), 
#                          labels = c(0, 1, 2))
# dataset$Purchased = factor(dataset$Purchased, 
#                            levels = c('No', 'Yes'), 
#                            labels = c(0, 1))

#Splitting data set into training set and test set
#install.packages('caTools')
# library(caTools)
# set.seed(123)
# split = sample.split(dataset$Purchased, SplitRatio = 0.8)
# training_set = subset(dataset, split== TRUE)
# test_set = subset(dataset, split== FALSE)

#Feature Scaling
#training_set[, 2:3] = scale(training_set[, 2:3])
#test_set[, 2:3] = scale(test_set[, 2:3])

#Fitting Support Vector Regression model to the dataset
#Create your regressor here
install.packages('e1071')
library(e1071)
regressor = svm(formula = Salary ~ ., data = dataset,
                type = 'eps-regression')

#Predicting a new result with
y_pred = predict(regressor, data.frame(Level = 6.5))
                
#Visualizing the linear regression results
library(ggplot2)
#For High Resolution Curves
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') + 
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)), 
              color = 'blue') + 
              ggtitle('Truth or Bluff (SVR)') + xlab('Level') + ylab('Salary')

File: support_vector_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sat Jan  7 20:10:17 2017

@author: AmoolyaD
"""

#SVR
#Data Preprocessing
#Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2].values

#Splitting the dataset into training set and test set
"""from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, random_state = 0)
"""
#feature scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_Y = StandardScaler()
X = sc_X.fit_transform(X)
Y = sc_Y.fit_transform(Y)


#Fitting SVR to the dataset
from sklearn.svm import SVR
regressor = SVR(kernel = 'rbf')
regressor.fit(X, Y)


#Predicting a new result with polynomial regression
y_pred = sc_Y.inverse_transform(regressor.predict(sc_X.transform(np.array([[6.5]]))))


#Visaulizing the Regression Results
plt.scatter(X, Y, color = 'red')
plt.plot(X, regressor.predict(X), color = 'blue')
plt.title('Truth of Bluff(SVR)')
plt.xlabel('Position Level')
plt.ylabel('Salary')

"""#Visaulizing the Regression Results for higher resolution
X_grid = np.arrange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, Y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.title('Truth of Bluff')
plt.xlabel('Position Level')
plt.ylabel('Salary')"""
